google/cloud/bigquery/v2/minimal/internal/dataset.h

```diff
@@ -0,0 +1,195 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_BIGQUERY_V2_MINIMAL_INTERNAL_DATASET_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_BIGQUERY_V2_MINIMAL_INTERNAL_DATASET_H
+
+#include "google/cloud/version.h"
+#include <nlohmann/json.hpp>
+#include <chrono>
+#include <memory>
+#include <string>
+
+namespace google {
+namespace cloud {
+namespace bigquery_v2_minimal_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+// Disabling clang-tidy here as the namespace is needed for using the
+// NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE_WITH_DEFAULT.
+using namespace nlohmann::literals;  // NOLINT
+
+auto const kDefaultTimepoint =
+    std::chrono::system_clock::time_point{std::chrono::milliseconds(0)};
```

Comment:
This is odd... 

https://en.cppreference.com/w/cpp/chrono/time_point/time_point

The default constructor of `std::chrono::system_clock::time_point` does exactly what you are doing here.  Why do we need a new symbol for it?